agent/http.go

```diff
@@ -359,14 +359,17 @@ func parseWait(resp http.ResponseWriter, req *http.Request, b *structs.QueryOpti
 
 // parseConsistency is used to parse the ?stale and ?consistent query params.
 // Returns true on error
-func parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
+func (s *HTTPServer) parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
 	query := req.URL.Query()
 	if _, ok := query["stale"]; ok {
 		b.AllowStale = true
 	}
 	if _, ok := query["consistent"]; ok {
 		b.RequireConsistent = true
 	}
+	if !b.AllowStale && !b.RequireConsistent {
```

Comment:
Suggestion to rewrite to make this boolean logic more readable. You could remove this block and init b.allowStale before line 364 with a single line like 
```
b.allowStale = !b.RequireConsistent && s.agent.config.HttpAPIStaleByDefault
```
The if block in line 364 after that will check query["stale"] after that to override if necessary.  IMO that's easier to follow than what you have now.